Cannot prepare Python binding on macOS M2

```
$ cd bindings/python
$ python3 -m venv venv  
$ source venv/bin/activate
$ pip install 'maturin[patchelf]'

Collecting maturin[patchelf]
  Using cached maturin-1.4.0-py3-none-macosx_10_12_x86_64.macosx_11_0_arm64.macosx_10_12_universal2.whl.metadata (18 kB)
Collecting patchelf (from maturin[patchelf])
  Using cached patchelf-0.17.2.1.tar.gz (167 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Using cached maturin-1.4.0-py3-none-macosx_10_12_x86_64.macosx_11_0_arm64.macosx_10_12_universal2.whl (15.7 MB)
Building wheels for collected packages: patchelf
  Building wheel for patchelf (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for patchelf (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [177 lines of output]
      
      
      --------------------------------------------------------------------------------
      -- Trying 'Ninja' generator
      --------------------------------
      ---------------------------
      ----------------------
      -----------------
      ------------
      -------
      --
      CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
        Compatibility with CMake < 3.5 will be removed from a future version of
        CMake.
      
        Update the VERSION argument <min> value or use a ...<max> suffix to tell
        CMake that the project does not need compatibility with older versions.
      
      Not searching for unused variables given on the command line.
      
      -- The C compiler identification is AppleClang 14.0.3.14030022
      -- Detecting C compiler ABI info
      -- Detecting C compiler ABI info - done
      -- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
      -- Detecting C compile features
      -- Detecting C compile features - done
      -- The CXX compiler identification is AppleClang 14.0.3.14030022
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      -- Configuring done (0.3s)
      -- Generating done (0.0s)
      -- Build files have been written to: /private/var/folders/ts/160w4k_j5mv5bjk9rnjkzjpm0000gn/T/pip-install-vgjfscmt/patchelf_cf6ee230c75e4e909b6944ec0234ca39/_cmake_test_compile/build
      --
      -------
      ------------
      -----------------
      ----------------------
      ---------------------------
      --------------------------------
      -- Trying 'Ninja' generator - success
      --------------------------------------------------------------------------------
      
      Configuring Project
        Working directory:
          /private/var/folders/ts/160w4k_j5mv5bjk9rnjkzjpm0000gn/T/pip-install-vgjfscmt/patchelf_cf6ee230c75e4e909b6944ec0234ca39/_skbuild/macosx-13.0-arm64-3.11/cmake-build
        Command:
          /opt/homebrew/bin/cmake /private/var/folders/ts/160w4k_j5mv5bjk9rnjkzjpm0000gn/T/pip-install-vgjfscmt/patchelf_cf6ee230c75e4e909b6944ec0234ca39 -G Ninja --no-warn-unused-cli -DCMAKE_INSTALL_PREFIX:PATH=/private/var/folders/ts/160w4k_j5mv5bjk9rnjkzjpm0000gn/T/pip-install-vgjfscmt/patchelf_cf6ee230c75e4e909b6944ec0234ca39/_skbuild/macosx-13.0-arm64-3.11/cmake-install -DPYTHON_VERSION_STRING:STRING=3.11.6 -DSKBUILD:INTERNAL=TRUE -DCMAKE_MODULE_PATH:PATH=/private/var/folders/ts/160w4k_j5mv5bjk9rnjkzjpm0000gn/T/pip-build-env-9zv_2fdr/overlay/lib/python3.11/site-packages/skbuild/resources/cmake -DPYTHON_EXECUTABLE:PATH=/Users/tison/Labs/vropendal/apache-opendal-incubating-0.44.1-src/bindings/python/venv/bin/python3.11 -DPYTHON_INCLUDE_DIR:PATH=/opt/homebrew/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11 -DPYTHON_LIBRARY:PATH=/opt/homebrew/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/lib/libpython3.11.dylib -DPython_EXECUTABLE:PATH=/Users/tison/Labs/vropendal/apache-opendal-incubating-0.44.1-src/bindings/python/venv/bin/python3.11 -DPython_ROOT_DIR:PATH=/Users/tison/Labs/vropendal/apache-opendal-incubating-0.44.1-src/bindings/python/venv -DPython_FIND_REGISTRY:STRING=NEVER -DPython_INCLUDE_DIR:PATH=/opt/homebrew/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11 -DPython3_EXECUTABLE:PATH=/Users/tison/Labs/vropendal/apache-opendal-incubating-0.44.1-src/bindings/python/venv/bin/python3.11 -DPython3_ROOT_DIR:PATH=/Users/tison/Labs/vropendal/apache-opendal-incubating-0.44.1-src/bindings/python/venv -DPython3_FIND_REGISTRY:STRING=NEVER -DPython3_INCLUDE_DIR:PATH=/opt/homebrew/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11 -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=13.0 -DCMAKE_OSX_ARCHITECTURES:STRING=arm64 -DCMAKE_BUILD_TYPE:STRING=Release
      
      Not searching for unused variables given on the command line.
      -- The C compiler identification is AppleClang 14.0.3.14030022
      -- The CXX compiler identification is AppleClang 14.0.3.14030022
      -- Detecting C compiler ABI info
      -- Detecting C compiler ABI info - done
      -- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
      -- Detecting C compile features
      -- Detecting C compile features - done
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      -- Configuring done (0.3s)
      -- Generating done (0.0s)
      -- Build files have been written to: /private/var/folders/ts/160w4k_j5mv5bjk9rnjkzjpm0000gn/T/pip-install-vgjfscmt/patchelf_cf6ee230c75e4e909b6944ec0234ca39/_skbuild/macosx-13.0-arm64-3.11/cmake-build
      [1/10] Creating directories for 'build_patchelf'
      [2/10] No download step for 'build_patchelf'
      [3/10] No update step for 'build_patchelf'
      [4/10] Performing patch step for 'build_patchelf'
      autoreconf: export WARNINGS=all
      autoreconf: Entering directory '.'
      autoreconf: configure.ac: not using Gettext
      autoreconf: running: aclocal --force
      autoreconf: configure.ac: tracing
      autoreconf: configure.ac: creating directory build-aux
      autoreconf: configure.ac: not using Libtool
      autoreconf: configure.ac: not using Intltool
      autoreconf: configure.ac: not using Gtkdoc
      autoreconf: running: /opt/homebrew/Cellar/autoconf/2.72/bin/autoconf --force
      autoreconf: configure.ac: not using Autoheader
      autoreconf: running: automake --add-missing --copy --force-missing
      configure.ac:14: installing 'build-aux/compile'
      configure.ac:5: installing 'build-aux/install-sh'
      configure.ac:5: installing 'build-aux/missing'
      src/Makefile.am: installing 'build-aux/depcomp'
      parallel-tests: installing 'build-aux/test-driver'
      autoreconf: 'build-aux/install-sh' is updated
      autoreconf: Leaving directory '.'
      [5/10] Performing configure step for 'build_patchelf'
      checking for a BSD-compatible install... /usr/bin/install -c
      checking whether build environment is sane... yes
      checking for a race-free mkdir -p... mkdir -p
      checking for gawk... no
      checking for mawk... no
      checking for nawk... no
      checking for awk... awk
      checking whether make sets $(MAKE)... yes
      checking whether make supports nested variables... yes
      checking for strip... strip
      checking for objdump... objdump
      checking for objcopy... no
      checking for readelf... no
      checking whether make supports the include directive... yes (GNU style)
      checking for gcc... gcc
      checking whether the C compiler works... yes
      checking for C compiler default output file name... a.out
      checking for suffix of executables...
      checking whether we are cross compiling... no
      checking for suffix of object files... o
      checking whether the compiler supports GNU C... yes
      checking whether gcc accepts -g... yes
      checking for gcc option to enable C11 features... none needed
      checking whether gcc understands -c and -o together... yes
      checking dependency style of gcc... gcc3
      checking for g++... g++
      checking whether the compiler supports GNU C++... yes
      checking whether g++ accepts -g... yes
      checking for g++ option to enable C++11 features... -std=gnu++11
      checking dependency style of g++ -std=gnu++11... gcc3
      checking dependency style of gcc... gcc3
      checking whether g++ -std=gnu++11 supports C++17 features with -std=c++17... yes
      checking that generated files are newer than configure... done
      configure: creating ./config.status
      config.status: creating Makefile
      config.status: creating src/Makefile
      config.status: creating tests/Makefile
      config.status: creating patchelf.spec
      config.status: executing depfiles commands
      [6/10] Performing build step for 'build_patchelf'
      FAILED: build_patchelf-prefix/src/build_patchelf-stamp/build_patchelf-build /private/var/folders/ts/160w4k_j5mv5bjk9rnjkzjpm0000gn/T/pip-install-vgjfscmt/patchelf_cf6ee230c75e4e909b6944ec0234ca39/_skbuild/macosx-13.0-arm64-3.11/cmake-build/build_patchelf-prefix/src/build_patchelf-stamp/build_patchelf-build
      cd /private/var/folders/ts/160w4k_j5mv5bjk9rnjkzjpm0000gn/T/pip-install-vgjfscmt/patchelf_cf6ee230c75e4e909b6944ec0234ca39/src/patchelf-upstream && /usr/bin/make check && /opt/homebrew/Cellar/cmake/3.28.1/bin/cmake -E touch /private/var/folders/ts/160w4k_j5mv5bjk9rnjkzjpm0000gn/T/pip-install-vgjfscmt/patchelf_cf6ee230c75e4e909b6944ec0234ca39/_skbuild/macosx-13.0-arm64-3.11/cmake-build/build_patchelf-prefix/src/build_patchelf-stamp/build_patchelf-build
      Making check in src
      g++ -std=gnu++11 -std=c++17 -DPACKAGE_NAME=\"patchelf\" -DPACKAGE_TARNAME=\"patchelf\" -DPACKAGE_VERSION=\"0.17.2\" -DPACKAGE_STRING=\"patchelf\ 0.17.2\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DPACKAGE=\"patchelf\" -DVERSION=\"0.17.2\" -DHAVE_CXX17=1 -I.    -Wall -std=c++17 -D_FILE_OFFSET_BITS=64     -g -O2 -MT patchelf.o -MD -MP -MF .deps/patchelf.Tpo -c -o patchelf.o patchelf.cc
      mv -f .deps/patchelf.Tpo .deps/patchelf.Po
      g++ -std=gnu++11 -std=c++17 -Wall -std=c++17 -D_FILE_OFFSET_BITS=64     -g -O2   -o patchelf patchelf.o
      Making check in tests
      /Applications/Xcode.app/Contents/Developer/usr/bin/make  simple-pie simple main too-many-strtab main-scoped big-dynstr no-rpath contiguous-note-sections libfoo.so libfoo-scoped.so libbar.so libbar-scoped.so libsimple.so libbuildid.so libtoomanystrtab.so phdr-corruption.so libbig-dynstr.debug
      gcc -DPACKAGE_NAME=\"patchelf\" -DPACKAGE_TARNAME=\"patchelf\" -DPACKAGE_VERSION=\"0.17.2\" -DPACKAGE_STRING=\"patchelf\ 0.17.2\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DPACKAGE=\"patchelf\" -DVERSION=\"0.17.2\" -DHAVE_CXX17=1 -I.    -fPIC -pie -g -O2 -MT simple_pie-simple.o -MD -MP -MF .deps/simple_pie-simple.Tpo -c -o simple_pie-simple.o `test -f 'simple.c' || echo './'`simple.c
      clang: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
      mv -f .deps/simple_pie-simple.Tpo .deps/simple_pie-simple.Po
      gcc -fPIC -pie -g -O2   -o simple-pie simple_pie-simple.o
      clang: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
      gcc -DPACKAGE_NAME=\"patchelf\" -DPACKAGE_TARNAME=\"patchelf\" -DPACKAGE_VERSION=\"0.17.2\" -DPACKAGE_STRING=\"patchelf\ 0.17.2\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DPACKAGE=\"patchelf\" -DVERSION=\"0.17.2\" -DHAVE_CXX17=1 -I.     -g -O2 -MT simple-simple.o -MD -MP -MF .deps/simple-simple.Tpo -c -o simple-simple.o `test -f 'simple.c' || echo './'`simple.c
      mv -f .deps/simple-simple.Tpo .deps/simple-simple.Po
      gcc  -g -O2   -o simple simple-simple.o
      gcc -DPACKAGE_NAME=\"patchelf\" -DPACKAGE_TARNAME=\"patchelf\" -DPACKAGE_VERSION=\"0.17.2\" -DPACKAGE_STRING=\"patchelf\ 0.17.2\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DPACKAGE=\"patchelf\" -DVERSION=\"0.17.2\" -DHAVE_CXX17=1 -I.    -fpic -g -O2 -MT main.o -MD -MP -MF .deps/main.Tpo -c -o main.o main.c
      mv -f .deps/main.Tpo .deps/main.Po
      gcc -DPACKAGE_NAME=\"patchelf\" -DPACKAGE_TARNAME=\"patchelf\" -DPACKAGE_VERSION=\"0.17.2\" -DPACKAGE_STRING=\"patchelf\ 0.17.2\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DPACKAGE=\"patchelf\" -DVERSION=\"0.17.2\" -DHAVE_CXX17=1 -I.    -fpic -g -O2 -MT foo.o -MD -MP -MF .deps/foo.Tpo -c -o foo.o foo.c
      foo.c:5:65: error: argument to 'section' attribute is not valid for this target: mach-o section specifier requires a segment and section separated by a comma
      const char __invoke_dynamic_linker__[] __attribute__ ((section (".interp"))) = "/foo/bar";
                                                                      ^
      1 error generated.
      make[2]: *** [foo.o] Error 1
      make[1]: *** [check-am] Error 2
      make: *** [check-recursive] Error 1
      ninja: build stopped: subcommand failed.
      Traceback (most recent call last):
        File "/private/var/folders/ts/160w4k_j5mv5bjk9rnjkzjpm0000gn/T/pip-build-env-9zv_2fdr/overlay/lib/python3.11/site-packages/skbuild/setuptools_wrap.py", line 674, in setup
          cmkr.make(make_args, install_target=cmake_install_target, env=env)
        File "/private/var/folders/ts/160w4k_j5mv5bjk9rnjkzjpm0000gn/T/pip-build-env-9zv_2fdr/overlay/lib/python3.11/site-packages/skbuild/cmaker.py", line 697, in make
          self.make_impl(clargs=clargs, config=config, source_dir=source_dir, install_target=install_target, env=env)
        File "/private/var/folders/ts/160w4k_j5mv5bjk9rnjkzjpm0000gn/T/pip-build-env-9zv_2fdr/overlay/lib/python3.11/site-packages/skbuild/cmaker.py", line 742, in make_impl
          raise SKBuildError(msg)
      
      An error occurred while building with CMake.
        Command:
          /opt/homebrew/bin/cmake --build . --target install --config Release --
        Install target:
          install
        Source directory:
          /private/var/folders/ts/160w4k_j5mv5bjk9rnjkzjpm0000gn/T/pip-install-vgjfscmt/patchelf_cf6ee230c75e4e909b6944ec0234ca39
        Working directory:
          /private/var/folders/ts/160w4k_j5mv5bjk9rnjkzjpm0000gn/T/pip-install-vgjfscmt/patchelf_cf6ee230c75e4e909b6944ec0234ca39/_skbuild/macosx-13.0-arm64-3.11/cmake-build
      Please check the install target is valid and see CMake's output for more information.
      
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for patchelf
Failed to build patchelf
ERROR: Could not build wheels for patchelf, which is required to install pyproject.toml-based projects

[notice] A new release of pip is available: 23.3.1 -> 23.3.2
[notice] To update, run: pip install --upgrade pip
```

cc @messense 

Although this seems like a maturin bug, I found it here and we have @messense maintain both projects :D

--- a/bindings/python/README.md
+++ b/bindings/python/README.md
@@ -82,7 +82,7 @@ source venv/bin/activate
 Install `maturin`:
 
 ```shell
-pip install maturin[patchelf]
+pip install maturin
 ```
 
 Build bindings:

